scripts/project_detection.sh

```diff
@@ -9,28 +9,62 @@
 # help detect which project (Mbed TLS, TF-PSA-Crypto)
 # or which Mbed TLS branch they are in.
 
+PROJECT_NAME_FILE="scripts/project_name.txt"
+
 # Project detection
 read_project_name_file () {
     SCRIPT_DIR=$(pwd)
 
-    PROJECT_NAME_FILE="scripts/project_name.txt"
-
     if read -r PROJECT_NAME < "$PROJECT_NAME_FILE"; then :; else
         echo "$PROJECT_NAME_FILE does not exist... Exiting..." >&2
         exit 1
     fi
 }
 
+# Check if the current folder is the Mbed TLS root one.
+#
+# Warning: if this is run from an non Mbed TLS/TF-PSA-Crypto root folder, the
+# script is terminated with a failure.
 in_mbedtls_repo () {
     read_project_name_file
     test "$PROJECT_NAME" = "Mbed TLS"
 }
 
+# Check if the current folder is the TF-PSA-Crypto root one.
+#
+# Warning: if this is run from an non Mbed TLS/TF-PSA-Crypto root folder, the
+# script is terminated with a failure.
 in_tf_psa_crypto_repo () {
     read_project_name_file
     test "$PROJECT_NAME" = "TF-PSA-Crypto"
 }
 
+# Check is $1 is either an Mbed TLS or TF-PSA-Crypto root folder.
+#
+# Differently from in_mbedtls_repo() and in_tf_psa_crypto_repo() above, this
+# can be run on any folder without causing the script to exit.
+is_valid_root () {
```

Comment:
Instead of `is_valid_root` and `check_root_type` as new utility functions having `is_mbedtls_root` and `is_tf_psa_crypto_root` would be better I think. Same as the related `in_mbedtls/tf_psa_crypto_root` but they take one path argument. Then `read_project_name_file` would handle zero or one path argument. In case of zero argument it takes as currently the current working directory. With these new utility functions we should be able to get rid of the IS_MBEDTLS_ROOT environment variable.

Reply:
Currently `read_project_name_file` fails if `PROJECT_NAME_FILE` is not found. This is something I avoided in `is_valid_root` so that I can call that function and prevent the calling script to exit while looking for the proper root.
I can however modify `read_project_name_file` and `in_mbedtls_repo`/`in_tf_psa_crypto_repo` in order to reduce the number of functions.